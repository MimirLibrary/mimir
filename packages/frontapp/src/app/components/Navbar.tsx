import React, { FC } from 'react';

const Navbar: FC = () => {
  return (
    <div>

    </div>
  );
};

export default Navbar;
