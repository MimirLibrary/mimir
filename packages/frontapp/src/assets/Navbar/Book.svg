<svg width="26" height="21" viewBox="0 0 28 21" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13 3.75C13 3.75 11.2 1.5 7 1.5C3.67639 1.5 1.85569 2.90898 1.24057 3.49699C1.07643 3.65389 1 3.87371 1 4.10078V17.6958C1 18.4812 1.92322 19.0659 2.68293 18.8665C3.65728 18.6108 5.07711 18.375 7 18.375C11.2 18.375 13 19.5 13 19.5M13 3.75V19.5M13 3.75C13 3.75 14.8 1.5 19 1.5C22.3236 1.5 24.1443 2.90898 24.7594 3.49699C24.9236 3.65389 25 3.87371 25 4.10078V17.6958C25 18.4812 24.0768 19.0659 23.3171 18.8665C22.3427 18.6108 20.9229 18.375 19 18.375C14.8 18.375 13 19.5 13 19.5" stroke="#333333" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>