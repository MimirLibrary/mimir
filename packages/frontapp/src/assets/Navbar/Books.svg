<svg width="32" height="32" viewBox="0 0 32 32" fill="current" xmlns="http://www.w3.org/2000/svg">
<mask id="path-1-inside-1_214_1282" fill="white">
<path d="M2 12L15.4592 3.34763C15.7887 3.13587 16.2114 3.13587 16.5408 3.34763L30 12H2Z"/>
</mask>
<path d="M2 12L0.918485 10.3176L-4.80963 14H2V12ZM30 12V14H36.8096L31.0815 10.3176L30 12ZM16.5408 3.34763L15.4592 5.02999L16.5408 3.34763ZM30 10H2V14H30V10ZM3.08152 13.6824L16.5408 5.02999L14.3777 1.66527L0.918485 10.3176L3.08152 13.6824ZM15.4592 5.02999L28.9185 13.6824L31.0815 10.3176L17.6223 1.66527L15.4592 5.02999ZM16.5408 5.02999C16.2114 5.24175 15.7887 5.24175 15.4592 5.02999L17.6223 1.66527C16.6341 1.02999 15.366 1.02999 14.3777 1.66527L16.5408 5.02999Z" mask="url(#path-1-inside-1_214_1282)"/>
<path d="M6 25C6 25.5523 6.44772 26 7 26C7.55228 26 8 25.5523 8 25H6ZM8 15C8 14.4477 7.55228 14 7 14C6.44772 14 6 14.4477 6 15H8ZM8 25V15H6V25H8Z"/>
<path d="M12 25C12 25.5523 12.4477 26 13 26C13.5523 26 14 25.5523 14 25H12ZM14 15C14 14.4477 13.5523 14 13 14C12.4477 14 12 14.4477 12 15H14ZM14 25V15H12V25H14Z" />
<path d="M18 25C18 25.5523 18.4477 26 19 26C19.5523 26 20 25.5523 20 25H18ZM20 15C20 14.4477 19.5523 14 19 14C18.4477 14 18 14.4477 18 15H20ZM20 25V15H18V25H20Z" />
<path d="M24 25C24 25.5523 24.4477 26 25 26C25.5523 26 26 25.5523 26 25H24ZM26 15C26 14.4477 25.5523 14 25 14C24.4477 14 24 14.4477 24 15H26ZM26 25V15H24V25H26Z"/>
<path d="M27 30C27.5523 30 28 29.5523 28 29C28 28.4477 27.5523 28 27 28L27 30ZM5 28C4.44772 28 4 28.4477 4 29C4 29.5523 4.44772 30 5 30L5 28ZM27 28L5 28L5 30L27 30L27 28Z" />
</svg>
