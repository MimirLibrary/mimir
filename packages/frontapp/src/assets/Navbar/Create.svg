<svg width="32" height="32" viewBox="0 0 32 32" fill="current" xmlns="http://www.w3.org/2000/svg">
<rect x="3" y="12" width="23" height="2" rx="1" />
<rect x="3" y="19" width="15" height="2" rx="1" />
<rect x="20" y="21" width="9" height="2" rx="1" />
<rect x="23.5" y="26.5" width="9" height="2" rx="1" transform="rotate(-90 23.5 26.5)"/>
<rect x="3" y="5" width="23" height="2" rx="1"/>
</svg>
