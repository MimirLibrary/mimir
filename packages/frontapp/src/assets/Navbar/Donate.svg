<svg width="26" height="26" viewBox="0 0 28 26" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M26 8.64386C26 7.75749 25.2724 7.03886 24.375 7.03886H20.6726C21.3904 6.33948 22.7784 4.52979 22.7784 3.40064C22.7784 2.01031 21.97 0.5 19.7007 0.5C16.7148 0.5 14.2204 4.53305 13.0483 6.29333C11.8755 4.53345 9.30475 0.500401 6.31922 0.500401C4.04991 0.500401 3.24147 2.01071 3.24147 3.40104C3.24147 4.52976 4.6421 6.33988 5.37457 7.03926H1.625C0.727594 7.03926 0 7.7579 0 8.64426V11.8092C0 12.7145 0.733903 13.4484 1.63922 13.4484V23.895C1.63922 24.7814 2.36679 25.5 3.26422 25.5H22.7561C23.6535 25.5 24.3811 24.7814 24.3811 23.895V13.448C25.2752 13.448 26 12.7232 26 11.8291L26 8.64386ZM19.7007 2.1054C20.7053 2.1054 21.1534 2.52233 21.1534 3.40107C21.1534 5.18382 19.1388 7.03845 17.2441 7.03845H14.4934C15.6479 5.34478 17.68 2.1054 19.7007 2.1054ZM6.31919 2.10543C8.34029 2.10543 10.4491 5.3448 11.6037 7.03808H8.85214C6.95739 7.03808 4.86642 5.15855 4.86642 3.37579C4.86642 2.49745 5.31451 2.10543 6.31919 2.10543ZM24.375 11.843H19.0937H13.8125V8.64386H24.375V11.843ZM1.62497 8.64386H12.1875V11.843H1.62497V8.64386ZM3.26419 13.448H12.1875V23.8946H3.26419V13.448ZM22.7565 23.8946H13.8125V13.448H18.2845H22.7565V23.8946Z" fill="#333333"/>
</svg>
