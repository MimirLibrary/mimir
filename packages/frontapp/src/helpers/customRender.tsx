import '@testing-library/jest-dom';
import { render, RenderOptions } from '@testing-library/react';
import React, { FC, ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import { ApolloProvider } from '@apollo/client';
import { client } from '@mimir/apollo-client';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string | null) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            return;
          }),
      },
    };
  },
}));

jest.unmock('react-redux');

const wrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <BrowserRouter>
      <ApolloProvider client={client}>
        <Provider store={store}>{children}</Provider>
      </ApolloProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => {
  return render(ui, { wrapper, ...options });
};

export { customRender as render };
