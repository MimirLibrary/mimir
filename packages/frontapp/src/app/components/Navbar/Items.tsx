import { ReactComponent as Home } from '../../../assets/Navbar/Home.svg';
import { ReactComponent as Library } from '../../../assets/Navbar/Library.svg';
import { ReactComponent as Book } from '../../../assets/Navbar/Book.svg';
import { ReactComponent as Donate } from '../../../assets/Navbar/Donate.svg';
import { ReactComponent as Settings } from '../../../assets/Navbar/Settings.svg';
import { ReactComponent as Readers } from '../../../assets/Navbar/Readers.svg';
import { ReactComponent as Books } from '../../../assets/Navbar/Books.svg';
import { ReactComponent as DonateFromUser } from '../../../assets/Navbar/DonateFromUser.svg';
import { ReactComponent as Create } from '../../../assets/Navbar/Create.svg';
import { ReactComponent as DonateToLibrary } from '../../../assets/Navbar/DonateToLibrary.svg';
import { ReactComponent as Logout } from '../../../assets/Navbar/Logout.svg';
import { RoutesTypes } from '../../../utils/routes';
import { NavbarItems } from '../../../utils/NavbarItems';

export const bottomNavSectionList = [
  {
    icon: <Settings />,
    name: NavbarItems.SETTINGS,
    path: RoutesTypes.SETTINGS,
  },
  {
    icon: <Logout />,
    name: NavbarItems.LOGOUT,
    path: '',
  },
];

export const readerNavList = [
  { icon: <Home />, name: NavbarItems.HOME, path: RoutesTypes.HOME },
  { icon: <Library />, name: NavbarItems.SEARCH, path: RoutesTypes.SEARCH },
  {
    icon: <DonateToLibrary />,
    name: NavbarItems.DONATE_TO_LIBRARY,
    path: RoutesTypes.DONATE_TO_LIBRARY,
  },
  {
    icon: <Book />,
    name: NavbarItems.HISTORY_OF_CLAIM,
    path: RoutesTypes.HISTORY_OF_CLAIM,
  },
  {
    icon: <Donate />,
    name: NavbarItems.HISTORY_OF_DONATE,
    path: RoutesTypes.HISTORY_OF_DONATE,
  },
];

export const managerNavList = [
  { icon: <Home />, name: NavbarItems.HOME, path: RoutesTypes.HOME },
  { icon: <Readers />, name: NavbarItems.READERS, path: RoutesTypes.READERS },
  {
    icon: <Books />,
    name: NavbarItems.BOOKS_STUFF,
    path: RoutesTypes.BOOKS_STUFF,
  },
  {
    icon: <DonateFromUser />,
    name: NavbarItems.DONATES_FROM_USER,
    path: RoutesTypes.DONATES_FROM_USER,
  },
  {
    icon: <Create />,
    name: NavbarItems.CREATE_NEW_ITEM,
    path: RoutesTypes.CREATE_NEW_ITEM,
  },
];
