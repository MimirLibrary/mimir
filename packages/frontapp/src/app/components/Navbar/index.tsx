import { FC } from 'react';
import NavbarItem from '../NavbarItem';
import styled from '@emotion/styled';
import { useAppDispatch } from '../../hooks/useTypedDispatch';
import { setActiveTab } from '../../store/slices/tabsSlice';
import { bottomNavSectionList, managerNavList, readerNavList } from './Items';
import { useAppSelector } from '../../hooks/useTypedSelector';
import { RolesTypes } from '@mimir/global-types';
import { colors, dimensions } from '@mimir/ui-kit';

interface IProps {
  hideSidebar: () => void;
}
const NavbarWrapper = styled.nav`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const NavbarTopSubsectionWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const NavbarBottomSubsectionWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: fixed;
  bottom: 0;

  @media (max-height: 747px) {
    position: static;
  }

  @media (max-width: ${dimensions.tablet_width}) {
    position: static;
  }
`;

const StyledHr = styled.hr`
  @media (max-width: ${dimensions.tablet_width}) {
    width: 100%;
    color: ${colors.gray_additional};
  }
`;

const Navbar: FC<IProps> = ({ hideSidebar }) => {
  const { userRole } = useAppSelector((state) => state.user);
  const dispatch = useAppDispatch();
  const changeActiveTab = (name: string) => {
    hideSidebar();
    dispatch(setActiveTab(name));
  };

  return (
    <NavbarWrapper data-testid="navbar">
      <NavbarTopSubsectionWrapper>
        {userRole === RolesTypes.READER
          ? readerNavList.map((item) => (
              <NavbarItem
                key={item.path}
                {...item}
                changeActiveTab={() => changeActiveTab(item.name)}
              />
            ))
          : managerNavList.map((item) => (
              <NavbarItem
                key={item.path}
                {...item}
                changeActiveTab={() => changeActiveTab(item.name)}
              />
            ))}
      </NavbarTopSubsectionWrapper>
      <StyledHr />
      <NavbarBottomSubsectionWrapper>
        {bottomNavSectionList.map((item) => (
          <NavbarItem
            key={item.path}
            {...item}
            changeActiveTab={() => changeActiveTab(item.name)}
          />
        ))}
      </NavbarBottomSubsectionWrapper>
    </NavbarWrapper>
  );
};

export default Navbar;
