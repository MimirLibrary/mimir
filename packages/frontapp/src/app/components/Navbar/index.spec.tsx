import { render } from '../../../helpers/customRender';
import { act, screen } from '@testing-library/react';
import * as reactHooks from 'react-redux';
import { RolesTypes } from '@mimir/global-types';
import Navbar from './index';
import { bottomNavSectionList, managerNavList, readerNavList } from './Items';

jest.mock('react-redux');
jest.mock('i18next', () => ({
  t: (str: any) => str,
}));

describe('render Navbar component', () => {
  const numberOfReaderItems =
    readerNavList.length + bottomNavSectionList.length;
  const numberOfManagerItems =
    managerNavList.length + bottomNavSectionList.length;

  it('should be correct reader navbar', async () => {
    jest.spyOn(reactHooks, 'useSelector').mockReturnValue(RolesTypes.READER);
    const handleChangeSomeData = jest.fn();
    render(<Navbar hideSidebar={handleChangeSomeData} />);
    expect(screen.getByTestId('navbar-active')).toHaveTextContent('Home');
    expect(screen.getAllByRole('link').length).toEqual(numberOfReaderItems);
  });

  it('should be correct manager navbar', async () => {
    jest.spyOn(reactHooks, 'useSelector').mockReturnValue(RolesTypes.MANAGER);
    const handleChangeSomeData = jest.fn();
    render(<Navbar hideSidebar={handleChangeSomeData} />);
    expect(screen.getByTestId('navbar-active')).toHaveTextContent('Home');
    expect(screen.getAllByRole('link').length).toEqual(numberOfManagerItems);
  });

  it('should completely match a snapshot', () => {
    jest.spyOn(reactHooks, 'useSelector').mockReturnValue(RolesTypes.READER);
    const handleChangeSomeData = jest.fn();
    render(<Navbar hideSidebar={handleChangeSomeData} />);
    expect(screen.getByTestId('navbar')).toMatchSnapshot();
  });

  it('should change active tab properly', () => {
    jest.spyOn(reactHooks, 'useSelector').mockReturnValue(RolesTypes.READER);
    const handleChangeSomeData = jest.fn();
    render(<Navbar hideSidebar={handleChangeSomeData} />);
    const libraryTab = document.querySelector('[data-testid="navbar-regular"]');
    expect(libraryTab).toHaveTextContent('Library');
    act(() => {
      libraryTab?.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(libraryTab).toHaveAttribute('primary', 'true');
  });
});
