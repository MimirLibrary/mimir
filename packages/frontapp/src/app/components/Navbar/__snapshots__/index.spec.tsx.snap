// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Navbar component should completely match a snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #1A1ED6;
  width: 15rem;
  height: 4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 6.25rem;
  padding-left: 1.5rem;
  margin-bottom: 0.12rem;
}

.emotion-2:hover {
  background: #1A1DB1;
  color: #FFFFFF;
}

.emotion-2:hover svg {
  -webkit-filter: invert(100%);
  filter: invert(100%);
}

.emotion-2:hover p {
  color: #FFFFFF;
}

.emotion-2:active {
  background: #0A0D95;
}

.emotion-2:active p {
  color: #FFFFFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 12.5rem;
  width: 100%;
  color: #FFFFFF;
}

.emotion-4 svg {
  margin-right: 1.25rem;
  width: 1.875rem;
  height: 2rem;
  -webkit-filter: invert(1);
  filter: invert(1);
}

.emotion-5 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  color: #FFFFFF;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  width: 15rem;
  height: 4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 6.25rem;
  padding-left: 1.5rem;
  margin-bottom: 0.12rem;
}

.emotion-6:hover {
  background: #1A1DB1;
  color: #FFFFFF;
}

.emotion-6:hover svg {
  -webkit-filter: invert(100%);
  filter: invert(100%);
}

.emotion-6:hover p {
  color: #FFFFFF;
}

.emotion-6:active {
  background: #0A0D95;
}

.emotion-6:active p {
  color: #FFFFFF;
}

.emotion-8 svg {
  margin-right: 1.25rem;
  width: 1.875rem;
  height: 2rem;
  -webkit-filter: invert(0);
  filter: invert(0);
}

.emotion-9 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  color: #333333;
}

@media (max-width: 48rem) {
  .emotion-22 {
    width: 100%;
    color: #E0E0E0;
  }
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  bottom: 0;
}

@media (max-height: 747px) {
  .emotion-23 {
    position: static;
  }
}

@media (max-width: 48rem) {
  .emotion-23 {
    position: static;
  }
}

<nav
  class="emotion-0"
  data-testid="navbar"
>
  <div
    class="emotion-0"
  >
    <a
      class="emotion-2"
      data-testid="navbar-active"
      href="/home"
      name="Home"
      primary="true"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <svg>
            Home.svg
          </svg>
        </div>
        <p
          class="emotion-5"
        >
          NavbarReader.Home
        </p>
      </div>
    </a>
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/search"
      name="Library"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            Library.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.Library
        </p>
      </div>
    </a>
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/donate-to-library"
      name="Donate to the library"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            DonateToLibrary.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.Donate to the library
        </p>
      </div>
    </a>
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/history-of-claim"
      name="History of claim"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            Book.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.History of claim
        </p>
      </div>
    </a>
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/history-of-donate"
      name="History of donate"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            Donate.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.History of donate
        </p>
      </div>
    </a>
  </div>
  <hr
    class="emotion-22"
  />
  <div
    class="emotion-23"
  >
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/settings"
      name="Settings"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            Settings.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.Settings
        </p>
      </div>
    </a>
    <a
      class="emotion-6"
      data-testid="navbar-regular"
      href="/"
      name="Log out"
      primary="false"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-8"
        >
          <svg>
            Logout.svg
          </svg>
        </div>
        <p
          class="emotion-9"
        >
          NavbarReader.Log out
        </p>
      </div>
    </a>
  </div>
</nav>
`;
